### PR TITLE
fix: Could not override GoldenTestTheme

### PR DESCRIPTION
**Pull request by Brainyoo**
## Description

All our tests failed because the default GoldenTestTheme changed the background colour. The GoldenTestTheme cannot be overwritten as it is not exported via the barrel file.

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [x] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Comment by Brainyoo** [comment]
We use AlchemistConfig.runWithConfig() for our tests. The background colour is applied with the changes. But at the time when the TableBorder is to be coloured, Zone.current[currentConfigKey] is suddenly null, which is why our colour is not applied.

File: lib/src/golden_test_runner.dart
Comment (btrautmann) [inline lib/src/golden_test_runner.dart:33]: Interesting that I didn't get a lint warning me that I had added a parameter to an overridden method that wasn't present in the paren't method 🤔

**Review by btrautmann — DISMISSED** [review]
Sorry about the inconvenience, thanks for catching this.

domain lgtm
platform lgtm

**Comment by btrautmann** [comment]
@Brainyoo it looks like there are a couple tests that need to be updated for these changes!

**Comment by btrautmann** [comment]
> We use AlchemistConfig.runWithConfig() for our tests. The background colour is applied with the changes. But at the time when the TableBorder is to be coloured, Zone.current[currentConfigKey] is suddenly null, which is why our colour is not applied.

@Brainyoo am I correct in saying that this would still exist after your PR lands, i.e this PR does _not_ address this bug?

File: lib/src/golden_test_theme.dart
Comment (btrautmann) [inline lib/src/golden_test_theme.dart:34]: Is there a benefit to making this a proper `ThemeExtension` or are we just wanting `copyWith` functionality?

**Review by btrautmann — COMMENTED** [review]
Just wondering about the Theme Extension and whether we're expecting consumers to leverage that meaningfully or if we just want copyWith which we can hand roll as we have elsewhere. I don't see lerp being used meaningfully which made me wonder.

**Review by btrautmann — DISMISSED** [review]
Ah OK, I got back on computer (was on mobile before) and now see how `ThemeExtension` is being used 👍 

domain lgtm
platform lgtm

**Comment by btrautmann** [comment]
The current failures are (partly) my fault, need to fix the job as it's failing to find the golden failures. Will open a PR to unblock this.

**Comment by btrautmann** [comment]
@Brainyoo looks like just one failing test for `borderColor`

**Review by btrautmann — APPROVED** [review]
domain lgtm
platform lgtm

**Comment by btrautmann** [comment]
Thank you @Brainyoo!

**Pull request merged**
